From ab7324eb90bab27fd83999ac31728dda2f04b045 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 21 Jun 2018 09:52:01 +0800
Subject: [PATCH 3999/5242] MLK-18649-1 arm64: fsl-imx8dx.dtsi: Correct
 power-domains for ldb nodes

commit  010cf4e808991af08cadf9909b12cc9481b6283e from
https://source.codeaurora.org/external/imx/linux-imx.git

The power domain devicetree binding documentation tells us that only
one phandle should be supplied to the power-domains property and if
multiple power domains are needed, then the power domains should be
chained as one parent power domain and multiple subsidiary power domains.
Thus, let's remove the unused power domain from the ldb nodes which
are originally expected to powerup the auxiliary MIPI/LVDS combo
subsystems.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit ebbd5aef7005c3128de215715448ac883ebc22e1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 2307ec6..51dcc10 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -1792,7 +1792,7 @@
 			 <&clk IMX8QXP_MIPI1_LVDS_PIXEL_CLK>,
 			 <&clk IMX8QXP_MIPI1_LVDS_BYPASS_CLK>;
 		clock-names = "pixel", "bypass", "aux_pixel", "aux_bypass";
-		power-domains = <&pd_mipi_dsi_0_lvds>, <&pd_mipi_dsi_1_lvds>;
+		power-domains = <&pd_mipi_dsi_0_lvds>;
 		gpr = <&lvds_region1>;
 		aux-gpr = <&lvds_region2>;
 		status = "disabled";
@@ -1968,7 +1968,7 @@
 			 <&clk IMX8QXP_MIPI0_LVDS_PIXEL_CLK>,
 			 <&clk IMX8QXP_MIPI0_LVDS_BYPASS_CLK>;
 		clock-names = "pixel", "bypass", "aux_pixel", "aux_bypass";
-		power-domains = <&pd_mipi_dsi_1_lvds>, <&pd_mipi_dsi_0_lvds>;
+		power-domains = <&pd_mipi_dsi_1_lvds>;
 		gpr = <&lvds_region2>;
 		aux-gpr = <&lvds_region1>;
 		status = "disabled";
-- 
1.7.9.5

